### PR TITLE
Modify lambda processor and sink documentation

### PR DESCRIPTION
**Pull request by srikanthjg**
### Description
Modify Lambda processor and sink documentation.
Splits the documentation into processor section and sink section.

### Issues Resolved
Closes #8655  

Dataprepper version 2.11

### Checklist
- [X ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
+title: aws_lambda
 parent: Processors
 grand_parent: Pipelines
 nav_order: 10
 ---
 
 # AWS Lambda integration for Data Prepper
 
-The AWS Lambda integration allows developers to use serverless computing capabilities within their Data Prepper pipelines for flexible event processing and data routing.
+The [AWS Lambda](https://aws.amazon.com/lambda/) integration allows developers to use serverless computing
```
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:11]:
```suggestion
The [AWS Lambda](https://aws.amazon.com/lambda/) integration allows developers to use serverless computing capabilities within their Data Prepper pipelines for flexible event processing and data routing.
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
 parent: Processors
 grand_parent: Pipelines
 nav_order: 10
 ---
 
 # AWS Lambda integration for Data Prepper
 
-The AWS Lambda integration allows developers to use serverless computing capabilities within their Data Prepper pipelines for flexible event processing and data routing.
+The [AWS Lambda](https://aws.amazon.com/lambda/) integration allows developers to use serverless computing capabilities within their Data Prepper pipelines for flexible event processing and data routing.
+capabilities within their Data Prepper pipelines for flexible event processing and data routing.
```
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:12]:
```suggestion
```

File: _data-prepper/pipelines/configuration/sinks/aws-lambda.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+```
+{% include copy-curl.html %}
+
+## Usage
+
+The invocation types are as follows:
+
+- `event` (Default): Executes functions asynchronously without waiting for responses.  
+- `request-response` (Sink only): Executes functions synchronously, though responses are not processed.
+- `batch`: Automatically groups events based on confirgured thresholds.
```
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/sinks/aws-lambda.md:60]:
```suggestion
- `batch`: Automatically groups events based on configured thresholds. 
```

File: _data-prepper/pipelines/configuration/processors/aws-lambda.md [hunk 4]
```diff
@@ -1,18 +1,18 @@
 ---
 layout: default
-title: AWS Lambda integration for Data Prepper
+title: aws_lambda
 parent: Processors
 grand_parent: Pipelines
 nav_order: 10
 ---
 
 # AWS Lambda integration for Data Prepper
```
Comment (vagimeli) [inline _data-prepper/pipelines/configuration/processors/aws-lambda.md:9]:
```suggestion
# aws_lambda integration for Data Prepper
```

File: _data-prepper/pipelines/configuration/sinks/aws-lambda.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+Field             | Type    | Required | Description                                                                 
+--------------------| ------- | -------- | ---------------------------------------------------------------------------- 
+`function_name`     | String  | Yes      | The name of the AWS Lambda function to invoke.                               
+`invocation_type`   | String  | No       | Specifies the invocation type. Default is `event`.             
+`aws.region`        | String  | Yes      | The AWS Region where the Lambda function is located.                         
+`aws.sts_role_arn`  | String  | No       | The ARN of the role to assume before invoking the Lambda function.               
+`max_retries`       | Integer | No       | The maximum number of retries if the invocation fails. Default is `3`.             
+`batch`             | Object  | No       | Optional batch settings for Lambda invocations. Default is `key_name = events`. Default threshold is `event_count=100`, `maximum_size="5mb"`, and `event_collect_timeout = 10s`.
+`lambda_when`       | String  | No       | Conditional expression to determine when to invoke the Lambda sink.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/sinks/aws-lambda.md:26]:
```suggestion
`lambda_when`       | String  | No       | A conditional expression that determines when to invoke the Lambda sink.          
```

File: _data-prepper/pipelines/configuration/sinks/aws-lambda.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+--------------------| ------- | -------- | ---------------------------------------------------------------------------- 
+`function_name`     | String  | Yes      | The name of the AWS Lambda function to invoke.                               
+`invocation_type`   | String  | No       | Specifies the invocation type. Default is `event`.             
+`aws.region`        | String  | Yes      | The AWS Region where the Lambda function is located.                         
+`aws.sts_role_arn`  | String  | No       | The ARN of the role to assume before invoking the Lambda function.               
+`max_retries`       | Integer | No       | The maximum number of retries if the invocation fails. Default is `3`.             
+`batch`             | Object  | No       | Optional batch settings for Lambda invocations. Default is `key_name = events`. Default threshold is `event_count=100`, `maximum_size="5mb"`, and `event_collect_timeout = 10s`.
+`lambda_when`       | String  | No       | Conditional expression to determine when to invoke the Lambda sink.          
+`dlq`               | Object  | No       | Dead-letter queue (DLQ) configuration for failed invocations.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/sinks/aws-lambda.md:27]:
```suggestion
`dlq`               | Object  | No       | The dead-letter queue (DLQ) configuration for failed invocations.                
```

File: _data-prepper/pipelines/configuration/sinks/aws-lambda.md [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+{% include copy-curl.html %}
+
+## Usage
+
+The invocation types are as follows:
+
+- `event` (Default): Executes functions asynchronously without waiting for responses.  
+- `request-response` (Sink only): Executes functions synchronously, though responses are not processed.
+- `batch`: Automatically groups events based on configured thresholds. 
+- `dlq`: Supports DLQ configuration for failed invocations after retry attempts.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/sinks/aws-lambda.md:61]:
```suggestion
- `dlq`: Supports the DLQ configuration for failed invocations after retry attempts.
```

File: _data-prepper/pipelines/configuration/sinks/aws-lambda.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+Data Prepper components use IAM role assumption, `aws.sts_role_arn`, for secure Lambda function invocation and respect Lambda's concurrency limits during event processing. For details, see the [AWS Lambda documentation](https://docs.aws.amazon.com/lambda).
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/sinks/aws-lambda.md:63]:
```suggestion
Data Prepper components use an AWS Identity and Access Management (IAM) role assumption, `aws.sts_role_arn`, for secure Lambda function invocation and respect Lambda's concurrency limits during event processing. For more information, see the [AWS Lambda documentation](https://docs.aws.amazon.com/lambda).
```

**Review by natebower — COMMENTED** [review]
@vagimeli @srikanthjg Please see my changes and let me know if you have any questions. Thanks!

**Review by vagimeli — APPROVED** [review]
Doc and editorial reviews completed. Editorial review feedback accepted and addressed.

**Pull request merged**